libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationV2Router.kt

```diff
@@ -0,0 +1,39 @@
+package com.mapbox.navigation.base.route
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.navigation.base.CurrentIndicesSnapshot
+
+/**
+ * Extends [NavigationRouter] to also provide ability for refreshing routes partially
+ * using indices stored in [CurrentIndicesSnapshot].
+ */
+interface NavigationV2Router : NavigationRouter {
```

Comment:
@mapbox/navigation-android , I like this name, but we haven't used similar name convention before. What do you think about it?

Reply:
It's just something off the top of my head. I'm open for suggestions. :)

Reply:
what if we add this interface unrer internal package for the very first interation and see how it's going

Reply:
So that we can break API later? Yeah, makes sense.

Reply:
yes, this way we don't expose the API and can break it 👍